src/staticanalysis/bot/static_analysis_bot/revisions.py

```diff
@@ -107,6 +107,17 @@ def __init__(self, description, api):
         revision = self.api.load_revision(self.phid)
         self.id = revision['id']
 
+        # Lookup base revision in refs
+        try:
+            refs = {
+                ref['type']: ref
+                for ref in diff['fields']['refs']
+            }
+            self.hg_base = refs['base']['identifier']
+        except KeyError:
```

Comment:
Shouldn't it always be available?

Reply:
No idea, it was not always the case on my initial tests (~ a year ago)

Reply:
Let's assume it always exists now (they made a lot of changes over the past months), asserting here (so we will get the error on Sentry and see if it is not actually the case).